.github/workflows/node-gyp-workaround.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
```

Comment:
maybe we could just do `curl -sSfLO https://raw.githubusercontent.com/mongodb-js/compass/42e6142ae08be6fec944b80ff6289e6bcd11badf/.evergreen/node-gyp-bug-workaround.sh && bash node-gyp-bug-workaround.sh` to use the script from Compass as a single source of truth? hopefully this isn’t something that we have to maintain in the long run, but … who knows 🥴

Reply:
There was a `node-gyp` release https://github.com/nodejs/node-gyp/releases 13 hours ago, might be getting solved soon. 🤞